src/Metadata/Property/Factory/ConstructorPropertyMetadataFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
```

Comment:
Do we really need this class? Can't we use https://github.com/symfony/symfony/pull/25605 directly?
https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/PropertyInfo/Metadata/Property/PropertyInfoPropertyMetadataFactory.php#L67 will do the job

Reply:
You're suggesting me to add things non-related to the property metadata factory inside of it. I don't think it's a great idea (it's about single responsibility).

It also enforces this feature to be inside the Symfony bridge (this is not a huge issue for me but all this work of decoupling for finally add things that may be decoupled easily from Symfony sounds sad to me).

Reply:
I’m not sure to understand your last comment. To clarify: if we drop this class, and bump PropertyInfo to 4.1 it will work without further modification right?

Reply:
Yes it should, @Nek- we just need to enforce `property-info` to `4.1` and it should take care of handling the constructor things for us.

Reply:
We only have to enforce it in our dev deps, with an older version, constructor support will not work but that's all.

Reply:
https://github.com/symfony/symfony/pull/25605 is about getting the type of a property via the constructor. I don't understand why you are speaking about it like it could replace the code of this class. Can you tell me more ?